### PR TITLE
chore: add job to check if all Go files have a license header

### PR DESCRIPTION
**Pull request by efekarakus**
_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: scripts/license.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    return 1
+}
+
+if [ $# != 1 ]; then
+    echo "Usage: $0 rootdir" >&2
+    exit 1
+fi
+
+fail=0
+for file in $(find $1 -name '*.go'); do
```
Comment (iamhopaul123) [inline scripts/license.sh]: Do we intend to only check for "*.go" files?

File: scripts/license.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    case $file in
+        $1/*/mocks/*)
+            # Skip mocks
+        ;;
+        *)
+            header="$(head -10 $file)"
```
Comment (iamhopaul123) [inline scripts/license.sh:38]: Just a nit but maybe we can be more aggressive about this number, since it doesn't hurt?

**Review by iamhopaul123 — COMMENTED** [review]
Looks awesome! Just a question and a nit.

File: scripts/license.sh (hunk 1 above)
Comment (efekarakus) [reply scripts/license.sh]: Updated also for our JS files

File: scripts/license.sh (hunk 2 above)
Comment (efekarakus) [reply scripts/license.sh:38]: Kept it as is, we can tweak the number if the checker ever fails

**Review by iamhopaul123 — APPROVED** [review]
:shipit:

File: scripts/license.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        ;;
+        $1/*/node_modules/*)
+            # Skip node modules for js files.
+        ;;
+        $1/site/*)
+            # Skip website content
+        ;;
+        *)
+            header="$(head -10 $file)"
+            if ! check_header "$header"; then
```
Comment (bvtujo) [inline scripts/license.sh:39]: Sorry, forgive my bash ignorance--why is this a different syntax than [L48](https://github.com/aws/copilot-cli/pull/1310/files/5a8e821eb61d4a11ff1ffc64a581415252f79767..3974fbfbc373d5c03468f0049400713c115c03ae#diff-3016f5e81e1b55c1248aa79ba8a929e8R48)?

**Review by bvtujo — APPROVED** [review]

File: scripts/license.sh (hunk 3 above)
Comment (iamhopaul123) [reply scripts/license.sh:39]: I guess `! check_header` means if the function doesn't exit successfully instead of directly comparing the return value?

Comment (efekarakus) [reply scripts/license.sh:39]: I don't know anything about bash either :P

From my understanding: https://stackoverflow.com/a/47876317/1201381 the first one runs the `check_header` function, if I want to run it then I can't use the test command `[` for it

**Pull request merged**